### PR TITLE
feat: add log rotate

### LINKED ISSUE #950: feat request: rotate log
### Issue description

当日志输出到文件时增加日志切割功能。
比如在我的环境，有一个自动删除3天没有写入日志的文件，ingress-controller是candidate状态的时候，就算是开启debug级别的日志也不会有日志输出，所以当过了3天日志文件就被清理了。之后ingress-controller成为了leader, 不过文件句柄还在，ingress-controller程序也能正常工作写入日志，但是日志采集程序就采不到日志了。希望能增加日志切割功能或定时写点日志也行。谢谢。

### Environment

- your apisix-ingress-controller version (output of apisix-ingress-controller version --long):
Version: 1.4.0
Git SHA: 1159522
Go Version: go1.17.2
Building OS/Arch: linux/amd64
Running OS/Arch: linux/amd64
- your Kubernetes cluster version (output of kubectl version):
- if you run apisix-ingress-controller in Bare-metal environment, also show your OS version (uname -a):

### PR DESCRIPTION
**Pull request by xiangtianyu**
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

add log rotate support
fixes #950

### CODE REVIEW

**Comment by tao12345666333** [comment]
sorry for delay, thanks for your contribution!

**Review by lingsamuel — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ In next release(v1.5), custom resource's API version v2 will be GA released. Ple
 * fix: ingress update event handler not filter by watching namespaces [#947](https://github.com/apache/apisix-ingress-controller/pull/)
 
 Please try out the release binaries and report any issues at
-https://github.com/apache/apisix-ingress-controller/issues.
+<https://github.com/apache/apisix-ingress-controller/issues>.
```
Comment (tao12345666333) [inline CHANGELOG.md:63]: CHANGELOG is automatically generated, I recommend you to unmodify this file.

File: pkg/config/config.go [hunk 2]
```diff
@@ -79,6 +79,10 @@ type Config struct {
 	KeyFilePath                string             `json:"key_file" yaml:"key_file"`
 	LogLevel                   string             `json:"log_level" yaml:"log_level"`
 	LogOutput                  string             `json:"log_output" yaml:"log_output"`
+	RotateOutputPath           string             `json:"rotate_output_path" yaml:"rotate_output_path"`
+	RotationMaxSize            int                `json:"rotation_max_size" yaml:"rotation_max_size"`
+	RotationMaxAge             int                `json:"rotation_max_age" yaml:"rotation_max_age"`
+	RotationMaxBackups         int                `json:"rotation_max_backups" yaml:"rotation_max_backups"`
```
Comment (tao12345666333) [inline pkg/config/config.go:85]: I suggest adding a LOG prefix to make it clear that it is related to log.

e.g. `LogRotationMaxSize`

Comment (xiangtianyu) [reply pkg/config/config.go]: I submit the commit to change the name, but there's dead link so i failed my test.How can i skip these checks?

Comment (tao12345666333) [reply pkg/config/config.go]: let me fix it.

Comment (tao12345666333) [reply pkg/config/config.go]: #1239 has been merge. the dead link has been fixed.

Comment (xiangtianyu) [reply pkg/config/config.go]: I got it. I've merged and pushed

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Pull request merged**